changelog.d/20230113_121808_kevin_fail_count_then_bail.rst

```diff
@@ -0,0 +1,5 @@
+New Functionality
+^^^^^^^^^^^^^^^^^
+
+- The funcX Endpoint will now shutdown -- and not continuously retry -- after 5
+  consecutive failures to successfully operate for at least 30s.
```

Comment:
30s is the heartbeat value.  It's somewhat arbitrary, but we don't currently _really_ have a heartbeat action.  We do a `log.debug()` every `heart_period` seconds (see near the end of `_main_loop()`.  We also no longer _register_ with the service every iteration.  Per #1006, we now register exactly once -- successfully or death -- at startup, and pass those credentials in to this class.  Meanwhile, I'm hesitant to say "register with web-service," or "connect to the AMQP service" because that may not be the only reason for failure.  This fail and bail approach is the general catch-all for exceptions.

So ... if the above is awkward, I'll see if I can draft it better.  I may reach out directly to work through this.

Reply:
Reworked.  Thanks!